src/peft/peft_model.py

```diff
@@ -2068,15 +2076,18 @@ def prepare_inputs_for_generation(self, *args, task_ids: Optional[torch.Tensor]
                 )
                 kwargs["token_type_ids"] = None
 
+            cache: transformers.Cache | None = model_kwargs.get("past_key_values", None)
```

Comment:
```suggestion
            cache: Optional[transformers.Cache] = model_kwargs.get("past_key_values", None)
```

since the `|` syntax is not compatible with python 3.9 (or we need a future import)